### DISCUSSION TITLE
Cmsis 5

### DISCUSSION THREAD

**Discussion by perigoso** [hathach/tinyusb#705] [General]
What was the reasoning to add the CMSIS_5 submodule locally?

Given that tinyusb is itself usually a submodule, this is not ideal, in my project I will effectively clone CMSIS_5 twice

**Comment by hathach** [comment]
Ah, you are right. I was optimizing ci build, and notice CMSIS_5 is rather huge (180 MB) and decided to include it locally to save clone time. Apparently it is not idea for super-module. I will do a revert, the github clone is pretty fast enough though.

**Reply by perigoso** [reply]
Ah, I see.

It's not exactly a problem, but I think it makes sense to keep it separate.

I started a "project", to host cmsis on github, since they are usually scatered around the web on vendor websites and they make sense as submodules, maybe you want to pull from there, it only contains the releases so it should be smaller.

https://github.com/cmsis-packs/cmsis-5

**Reply by perigoso** [reply]
Hm, just pulled it out of curiosity and its not much smaller at 137.75 MiB

**Reply by hathach** [reply]
Yeah, the dsp, documentation, and example are a bit heavy. Though tinyusb doesn’t make use any of those. Anyway, I have added cmsis5 back as submodules. Just update the ci and bsp to only checkout if needed.

**Reply by perigoso** [reply]
I might consider not deploying documentation on those repos, right now the scripts just port pretty much everything, also considering splitting cmsis-5 into its various components
